pkg/integration/handler.go

```diff
@@ -0,0 +1,214 @@
+package integration
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/dhis2-sre/im-manager/pkg/config"
+
+	"github.com/dhis2-sre/im-manager/internal/handler"
+	"github.com/gin-gonic/gin"
+)
+
+func NewHandler(config config.Config, client DockerHubClient) Handler {
+	return Handler{config, client}
+}
+
+type Handler struct {
+	config          config.Config
+	dockerHubClient DockerHubClient
+}
+
+type DockerHubClient interface {
+	GetImages(organization string) ([]string, error)
+	GetTags(organization, repository string) ([]string, error)
+}
+
+type Request struct {
+	Key     string `json:"key" binding:"required"`
+	Payload any    `json:"payload"`
+}
+
+// Integrations ...
+// swagger:route POST /integrations postIntegration
+//
+// Return integration for a given key
+//
+// Security:
+//  oauth2:
+//
+// responses:
+//   200: Any
+//   401: Error
+//   403: Error
+//   415: Error
+func (h Handler) Integrations(c *gin.Context) {
+	var request Request
+	if err := handler.DataBinder(c, &request); err != nil {
+		_ = c.Error(err)
+		return
+	}
+
+	if request.Key == "IMAGE_REPOSITORY" {
+		payload := request.Payload.(map[string]any)
+		organization := payload["organization"].(string)
+		images, err := h.dockerHubClient.GetImages(organization)
+		if err != nil {
+			_ = c.Error(err)
+			return
+		}
+
+		c.JSON(http.StatusOK, images)
+		return
+	}
+
+	if request.Key == "IMAGE_TAG" {
+		payload := request.Payload.(map[string]any)
+		organization := payload["organization"].(string)
+		repository := payload["repository"].(string)
+		tags, err := h.dockerHubClient.GetTags(organization, repository)
+		if err != nil {
+			_ = c.Error(err)
+			return
+		}
+
+		c.JSON(http.StatusOK, tags)
+		return
+	}
+
+	if request.Key == "DATABASE_ID" {
+		token, err := handler.GetTokenFromHttpAuthHeader(c)
+		if err != nil {
+			_ = c.Error(err)
+			return
+		}
+
+		url := fmt.Sprintf("http://%s/databases", h.config.DatabaseManagerService.Host)
+		databases, err := getDatabases(token, url)
+		if err != nil {
+			_ = c.Error(err)
+			return
+		}
+
+		c.JSON(http.StatusOK, databases)
+		return
+	}
+
+	if request.Key == "PRESET_ID" {
+		token, err := handler.GetTokenFromHttpAuthHeader(c)
+		if err != nil {
+			_ = c.Error(err)
+			return
+		}
+
+		url := fmt.Sprintf("http://%s/presets", h.config.InstanceService.Host)
+		presets, err := getInstances(token, url)
+		if err != nil {
+			_ = c.Error(err)
+			return
+		}
+
+		c.JSON(http.StatusOK, presets)
+		return
+	}
+
+	if request.Key == "SOURCE_ID" {
+		token, err := handler.GetTokenFromHttpAuthHeader(c)
+		if err != nil {
+			_ = c.Error(err)
+			return
+		}
+
+		url := fmt.Sprintf("http://%s/instances", h.config.InstanceService.Host)
+		presets, err := getInstances(token, url)
+		if err != nil {
+			_ = c.Error(err)
+			return
+		}
+
+		c.JSON(http.StatusOK, presets)
+		return
+	}
+
+	c.Status(http.StatusNotFound)
+}
+
+func getInstances(token string, url string) (map[uint]string, error) {
```

Comment:
I would move this below declaration of getDatabases so they are declared in the order they are used.